canvas_sdk/questionnaires/utils.py

```diff
@@ -0,0 +1,122 @@
+import functools
+import json
+from collections.abc import Generator
+from pathlib import Path
+from typing import Any, TypedDict
+
+import yaml
+from jsonschema import Draft7Validator, validators
+
+from canvas_sdk.utils.plugins import plugin_only
+
+
+class Response(TypedDict):
+    """A Response of a Questionnaire."""
+
+    name: str
+    code: str
+    code_description: str
+    value: str
+
+
+class Question(TypedDict):
+    """A Question of a Questionnaire."""
+
+    name: str
+    code_system: str
+    code: str
+    code_description: str
+    content: str
+    responses_code_system: str
+    responses_type: str
+    use_in_shx: bool
+    responses: list[Response]
+
+
+class QuestionnaireConfig(TypedDict):
+    """A Questionnaire configuration."""
+
+    name: str
+    use_case_in_charting: str
+    code_system: str
+    code: str
+    can_originate_in_charting: bool
+    search_tags: str
+    scoring_code_system: str
+    scoring_code: str
```

Comment:
I think these should be removed, there are a few other inconsistencies here (`use_in_shx` for example)